components/application-mgt/org.wso2.carbon.identity.application.common/src/main/java/org/wso2/carbon/identity/application/common/model/UserDefinedAuthenticatorEndpointConfig.java

```diff
@@ -32,6 +32,11 @@ public class UserDefinedAuthenticatorEndpointConfig {
 
     private final EndpointConfig endpointConfig;
 
+    public UserDefinedAuthenticatorEndpointConfig(EndpointConfig endpointConfig) {
+
+        this.endpointConfig = endpointConfig;
+    }
+
```

Comment:
why we need this ? Then why we need a builder and have a private constructor ?

Reply:
With this constructor introduced, we can create UserDefinedAuthenticatorEndpointConfig object just giving the endpoint object from action, like here https://github.com/wso2/carbon-identity-framework/pull/6105/files#diff-e15e35779957907ba8ea90f73d573eed6e57f3180c7412df385ec887597b20a7R128

Reply:
You can use the builder as well. Switch to one mode